test/standardTokenGateway.js

```diff
@@ -0,0 +1,28 @@
+const { ethers } = require('hardhat');
+const { expect } = require('chai');
+const { CHAINLINK_DEC, etherBalances, DEFAULT_CHAINLINK_EUR_USD_PRICE, defaultConvertUsdToEur } = require('./common');
+
+describe('StandardTokenGateway', async () => {
+  let StandardTokenGateway;
+  // $1.05
+  const eurUsdPrice = DEFAULT_CHAINLINK_EUR_USD_PRICE;
+
+  beforeEach(async () => {
+    const ChainlinkMock = await (await ethers.getContractFactory('Chainlink')).deploy(eurUsdPrice);
+    const TST = await (await ethers.getContractFactory('DUMMY')).deploy('Standard Token', 'TST', 18);
+    const SEuro = await (await ethers.getContractFactory('DUMMY')).deploy('SEuro', 'SEUR', 18);
+    StandardTokenGateway = await (await ethers.getContractFactory('StandardTokenGateway')).deploy(
+      TST.address, SEuro.address, ChainlinkMock.address, CHAINLINK_DEC
+    );
+  });
+
+  it('provides the TST exchange for sEURO and the other asset', async () => {
```

Comment:
good to create a new test file for gateway, can you add one with 6 decimals too (e.g, USDT)?

Reply:
this test suite probably won't exist if the calculations are moved to outside of this gateway contract

Reply:
that's alright, just so we test with other decimal values than the erc-20 default one (18)